src/kademlia/dht_storage.cpp

```diff
@@ -189,9 +201,8 @@ namespace
 
 	public:
 
-		dht_default_storage(sha1_hash const& id, dht_settings const& settings)
-			: m_id(id)
-			, m_settings(settings)
+		dht_default_storage(dht_settings const& settings)
+			: m_settings(settings)
 		{
 			memset(&m_counters, 0, sizeof(m_counters));
```

Comment:
unrelated to this review. this looks a bit suspicious. we don't own m_counters (it's a reference) but we still just erase it's storage. It should probably have a clear function on it instead that we can call here. If it's just a plain array, it should probably be a std::array<> that we can call fill() on here